standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -235,6 +235,10 @@ public void alterTable(RawStore msdb, Warehouse wh, String catName, String dbnam
 
       boolean renamedTranslatedToExternalTable = rename && MetaStoreUtils.isTranslatedToExternalTable(oldt)
           && MetaStoreUtils.isTranslatedToExternalTable(newt);
+
+      List<ColumnStatistics> columnStatistics = getColumnStats(msdb, oldt);
+      columnStatistics = deleteTableColumnStats(msdb, oldt, newt, columnStatistics);
```

Comment:
Since we have deleted the table column stats, do we need to call deleteAllPartitionColumnStatistics here?
[https://github.com/apache/hive/blob/f6bd0eb80767adfa9ce9f47a6d02a4940903effb/stand[…]ain/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java](https://github.com/apache/hive/blob/f6bd0eb80767adfa9ce9f47a6d02a4940903effb/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java#L417)

Reply:
deleteAllPartitionColumnStatistics is used only in one specific case: [HIVE-23959](https://issues.apache.org/jira/browse/HIVE-23959): Provide an option to wipe out column stats for partitioned tables in case of column removal

So I don't think we should call this method in all cases. However, it made me think that in this particular case the table column statistics should not be re-added at the end. What do you think?

Reply:
After discussing offline, deleteAllPartitionedColumnStats and deleteTableColumnStats seem to have different functionality altogether and this call seems necessary.